gravitee-apim-console-webui/src/management/api/api-traffic-v4/webhook-logs/utils/webhook-metrics.utils.ts

```diff
@@ -0,0 +1,180 @@
+ * limitations under the License.
+ */
+
+import { ConnectionLog } from '../../../../../entities/management-api-v2';
+import { WebhookAdditionalMetrics, WebhookLog } from '../models/webhook-logs.models';
+
+/**
+ * Safely parses a numeric string, returning 0 if the value is not a valid number.
+ */
+function parseNumberOrZero(raw?: string): number {
+  const num = Number(raw);
+  return Number.isNaN(num) ? 0 : num;
+}
+
+/**
+ * Safely parses a numeric string, returning undefined if the value is not a valid number.
+ */
+function parseNumberOrUndefined(raw?: string): number | undefined {
+  const num = Number(raw);
+  return Number.isNaN(num) ? undefined : num;
+}
+
+/**
+ * Extracts and transforms additionalMetrics from the backend format
+ * to the typed WebhookAdditionalMetrics format.
+ */
+export function extractWebhookAdditionalMetrics(additionalMetrics: { [key: string]: string }): Partial<WebhookAdditionalMetrics> {
+  const metrics: Partial<WebhookAdditionalMetrics> = {};
+
+  // Fields that are always present (per connector readme)
+  metrics['string_webhook_req-method'] = additionalMetrics['string_webhook_req-method'];
+  metrics['string_webhook_url'] = additionalMetrics['string_webhook_url'];
+  metrics['keyword_webhook_app-id'] = additionalMetrics['keyword_webhook_app-id'];
+  metrics['keyword_webhook_sub-id'] = additionalMetrics['keyword_webhook_sub-id'];
+  metrics['json_webhook_retry-timeline'] = additionalMetrics['json_webhook_retry-timeline'];
+
+  // These fields are supposed to be always present per connector docs,
+  // but when entrypoint logging is disabled, additionalMetrics is empty.
+  // Only set them if they exist in the source to avoid misleading default values.
+  if (additionalMetrics['long_webhook_req-timestamp'] !== undefined) {
+    metrics['long_webhook_req-timestamp'] = parseNumberOrZero(additionalMetrics['long_webhook_req-timestamp']);
+  }
+  if (additionalMetrics['int_webhook_resp-status'] !== undefined) {
+    metrics['int_webhook_resp-status'] = parseNumberOrZero(additionalMetrics['int_webhook_resp-status']);
+  }
+  if (additionalMetrics['int_webhook_retry-count'] !== undefined) {
+    metrics['int_webhook_retry-count'] = parseNumberOrZero(additionalMetrics['int_webhook_retry-count']);
+  }
+
+  // bool_webhook_dlq - only set if present in additionalMetrics
+  if (additionalMetrics['bool_webhook_dlq'] !== undefined) {
+    metrics['bool_webhook_dlq'] = additionalMetrics['bool_webhook_dlq'] === 'true';
+  }
+
+  // Conditional fields - only assign if present
+  if (additionalMetrics['string_webhook_last-error'] !== undefined) {
+    metrics['string_webhook_last-error'] = additionalMetrics['string_webhook_last-error'] || null;
+  }
+
+  if (additionalMetrics['json_webhook_req-headers'] !== undefined) {
+    metrics['json_webhook_req-headers'] = additionalMetrics['json_webhook_req-headers'] || null;
+  }
+
+  if (additionalMetrics['string_webhook_req-body'] !== undefined) {
```

Comment:
All this checked can be simplified : 
```suggestion
  if (additionalMetrics['string_webhook_req-body']) {
```

See https://www.learn-ts.org/en/Truthy_and_Falsy

Reply:
Had to completely rework this. Should be taken care of by refactoring.